frontend/src/components/Header/Header.tsx

```diff
@@ -44,32 +49,70 @@ export function Header(): JSX.Element {
 
   const [open, setOpen] = useState(false);
   const [listItems, setListItems] = useState([
-    { link: "/", text: t("homepage") }
+    { link: "/", text: t("homepage") },
   ]);
 
-  useEffect(() => {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLButtonElement>(
+    null,
+  );
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  React.useEffect(() => {
     const baseItems = [{ link: "/", text: t("homepage") }];
     const additionalItems = [
       { link: "/projects", text: t("myProjects") },
-      { link: "/courses", text: t("myCourses") }
+      { link: "/courses", text: t("myCourses") },
     ];
-    if (isLoggedIn()) {
+    if (me.role !== "UNKNOWN") {
       setListItems([...baseItems, ...additionalItems]);
-    }
-    else {
+    } else {
       setListItems(baseItems);
     }
-  }, [t]);
+  }, [me, t]);
 
   return (
     <Box sx={{ flexGrow: 1 }}>
       <AppBar position="sticky">
         <Toolbar disableGutters>
-          <IconButton edge="start" onClick={() => setOpen(!open)} sx={{ color: "white", marginLeft: 0 }}>
-            <MenuIcon style={{fontSize:"2rem"}} />
+          <IconButton
+            edge="start"
+            onClick={() => setOpen(!open)}
+            sx={{ color: "white", marginLeft: 0 }}
+          >
+            <MenuIcon style={{ fontSize: "2rem" }} />
           </IconButton>
-          <TitlePortal/>
-          <LoginButton></LoginButton>
+          <TitlePortal />
+          {me.role !== "UNKNOWN" && (
+            <>
+              <IconButton
+                edge="end"
+                onClick={handleClick}
+                sx={{ color: "inherit", marginRight: "0.3rem" }}
+              >
+                <AccountCircleIcon />
+                <Typography variant="body1" sx={{ paddingLeft: "0.3rem" }}>
+                  {me.display_name}
+                </Typography>
+              </IconButton>
+              <Menu
+                anchorEl={anchorEl}
+                open={Boolean(anchorEl)}
+                onClose={() => setAnchorEl(null)}
+              >
+                <Typography sx={{ padding: "6px 16px", color: "black" }}>
+                  {me.display_name}
+                </Typography>
+                <MenuItem
+                  onClick={() => (window.location.href = `${API_URL}/logout`)}
```

Comment:
Don't do `window.location.href` there are functions made for redirects, `navigate`

Reply:
not fixed